google/resource_compute_address.go

```diff
@@ -88,108 +113,266 @@ func resourceComputeAddress() *schema.Resource {
 	}
-	d.Set("subnetwork", ConvertSelfLinkToV1(addr.Subnetwork))
-	d.Set("address", addr.Address)
-	d.Set("self_link", ConvertSelfLinkToV1(addr.SelfLink))
-	d.Set("name", addr.Name)
-	d.Set("network_tier", addr.NetworkTier)
-	d.Set("project", addressId.Project)
-	d.Set("region", GetResourceNameFromSelfLink(addr.Region))
 
 	return nil
 }
 
 func resourceComputeAddressDelete(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	addressId, err := parseComputeAddressId(d.Id(), config)
+	project, err := getProject(d, config)
 	if err != nil {
 		return err
 	}
 
-	// Delete the address
-	op, err := config.clientComputeBeta.Addresses.Delete(
-		addressId.Project, addressId.Region, addressId.Name).Do()
+	url, err := replaceVars(d, config, "https://www.googleapis.com/compute/beta/projects/{{project}}/regions/{{region}}/addresses/{{name}}")
 	if err != nil {
-		return fmt.Errorf("Error deleting address: %s", err)
+		return err
 	}
 
-	err = computeSharedOperationWait(config.clientCompute, op, addressId.Project, "Deleting Address")
+	log.Printf("[DEBUG] Deleting Address %q", d.Id())
+	res, err := Delete(config, url)
+	if err != nil {
+		return handleNotFoundError(err, d, "Address")
+	}
+
+	op := &compute.Operation{}
+	err = Convert(res, op)
 	if err != nil {
 		return err
 	}
 
-	d.SetId("")
+	err = computeOperationWaitTime(
+		config.clientCompute, op, project, "Deleting Address",
+		int(d.Timeout(schema.TimeoutDelete).Minutes()))
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Finished deleting Address %q: %#v", d.Id(), res)
 	return nil
 }
 
-func resourceComputeAddressImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+func resourceComputeAddressImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	config := meta.(*Config)
+	parseImportId([]string{"projects/(?P<project>[^/]+)/regions/(?P<region>[^/]+)/addresses/(?P<name>[^/]+)", "(?P<project>[^/]+)/(?P<region>[^/]+)/(?P<name>[^/]+)", "(?P<name>[^/]+)"}, d, config)
 
-	addressId, err := parseComputeAddressId(d.Id(), config)
+	// Replace import id for the resource id
+	id, err := replaceVars(d, config, "{{project}}/{{region}}/{{name}}")
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Error constructing id: %s", err)
 	}
-
-	d.SetId(addressId.canonicalId())
+	d.SetId(id)
 
 	return []*schema.ResourceData{d}, nil
 }
+
+func flattenComputeAddressAddress(v interface{}) interface{} {
+	return v
+}
+
+func flattenComputeAddressAddressType(v interface{}) interface{} {
+	if v == nil || v.(string) == "" {
+		return "EXTERNAL"
+	}
+	return v
+}
+
+func flattenComputeAddressCreationTimestamp(v interface{}) interface{} {
+	return v
+}
+
+func flattenComputeAddressDescription(v interface{}) interface{} {
+	return v
+}
+
+func flattenComputeAddressName(v interface{}) interface{} {
+	return v
+}
+
+func flattenComputeAddressNetworkTier(v interface{}) interface{} {
```

Comment:
Question: have we ever thought about removing these no-op expanders and flatteners?  Seems like it'd be easy-ish to do and make the code a lot more fluent.

Reply:
Yeah, I remember @rosbo wanted to leave them in in the beginning, I think the rationale was just "it's generated code anyway, it doesn't really matter". I'm fine either way. Vincent, anything else to comment about it?